src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -69,15 +72,21 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
 
             try
             {
-                HttpResponse response = await _httpManager.SendGetAsync(_ImdsUri, Headers, logger).ConfigureAwait(false);
+                HttpResponse response = await _httpManager.SendGetAsync(ImdsUri, Headers, logger).ConfigureAwait(false);
+
+                if (response.StatusCode != HttpStatusCode.BadRequest)
+                {
```

Comment:
if the statusCode is anything but badRequest, you will send the request twice? and then in UpdateImdsUriApiVersionAsync there is another .SendGetAsync(). Can you explain what the issue is that you're fixing? that might help me understand better. thx.

Reply:
It was a typo before. In case the status is a BadRequest, that implies the version is outdated and not supported any longer.